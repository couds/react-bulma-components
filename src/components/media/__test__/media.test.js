import React from 'react';
import renderer from 'react-test-renderer';
import Media from '..';

describe('Media component', () => {
  it('Should exist', () => {
    expect(Media).toMatchSnapshot();
  });
  it('Should expose Level Side and Item', () => {
    expect(Media.Content).toMatchSnapshot();
    expect(Media.Item).toMatchSnapshot();
  });
  it('Should have media classname', () => {
    const component = renderer.create(
      <Media>
        <img
          alt="placeholder"
          src="http://bulma.io/images/placeholders/128x128.png"
        />
      </Media>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should hbe a Media Item', () => {
    const component = renderer.create(
      <Media.Item renderAs="figure" position="left">
        <img
          alt="placeholder"
          src="http://bulma.io/images/placeholders/128x128.png"
        />
      </Media.Item>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should hbe a Media Item Centered', () => {
    const component = renderer.create(
      <Media.Item renderAs="figure" position="center">
        <img
          alt="placeholder"
          src="http://bulma.io/images/placeholders/128x128.png"
        />
      </Media.Item>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be a Media Content', () => {
    const component = renderer.create(
      <Media.Content>
        <p>Lorem Ipsum</p>
      </Media.Content>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(
      <Media className="other-class this-is-a-test">
        <p>Default</p>
      </Media>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(
      <Media style={{ height: 250 }}>
        <p>Default</p>
      </Media>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
