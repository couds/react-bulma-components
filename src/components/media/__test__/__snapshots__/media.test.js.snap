// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media component Should be a Media Content 1`] = `
<div
  className="content"
  style={Object {}}
>
  <p>
    Lorem Ipsum
  </p>
</div>
`;

exports[`Media component Should concat classname in props with Bulma classname 1`] = `
<article
  className="media other-class this-is-a-test"
  style={Object {}}
>
  <p>
    Default
  </p>
</article>
`;

exports[`Media component Should exist 1`] = `[Function]`;

exports[`Media component Should expose Level Side and Item 1`] = `[Function]`;

exports[`Media component Should expose Level Side and Item 2`] = `[Function]`;

exports[`Media component Should have media classname 1`] = `
<article
  className="media"
  style={Object {}}
>
  <img
    alt="placeholder"
    src="http://bulma.io/images/placeholders/128x128.png"
  />
</article>
`;

exports[`Media component Should hbe a Media Item 1`] = `
<figure
  className="media-left"
  style={Object {}}
>
  <img
    alt="placeholder"
    src="http://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`Media component Should hbe a Media Item Centered 1`] = `
<figure
  className="media-content"
  style={Object {}}
>
  <img
    alt="placeholder"
    src="http://bulma.io/images/placeholders/128x128.png"
  />
</figure>
`;

exports[`Media component Should use inline styles 1`] = `
<article
  className="media"
  style={
    Object {
      "height": 250,
    }
  }
>
  <p>
    Default
  </p>
</article>
`;
